python/src/scipp/plotting/model2d.py

```diff
@@ -7,14 +7,29 @@
 from .model import PlotModel
 from .resampling_model import resampling_model
 from .._scipp import core as sc
+from .._utils.is_type import is_discrete
 import numpy as np
 
 
+def _check_if_dtypes_supported(scipp_obj_dict, *_args, **_kwargs):
+    for array_name, array in scipp_obj_dict.items():
+        if sc.is_bins(array):  # We fully support binned data.
+            return
+        for coord_name, coord in array.coords.items():
+            if is_discrete(coord):
+                raise NotImplementedError(
+                    "Plotting 2D data with discrete coordinates "
+                    "(e.g. int) is not supported. "
```

Comment:
Um, this sounds bad? We won't be able to plot any (non-event) neutron data, because we always have an integer "spectrum" coordinate in this case.

Reply:
Is binned data always fine? If so I could adjust the test and only forbid dense data.

Reply:
Binned data is not enough. We have 2-D dense (histogrammed) data with an integer coord.